src/ast/decl.rs

```diff
@@ -210,6 +210,7 @@ pub struct ProcDecl {
     /// (read) access to the proc declaration.
     pub body: RefCell<Option<Block>>,
     pub span: Span,
+    pub calculus: Option<Ident>,
```

Comment:
Why would this be an `Ident` if the calculus is always from a fixed set of values that are built into the compiler? Couldn't you save a value of an enum type `Calculus` here?